archunit/src/main/java/com/tngtech/archunit/library/plantuml/rules/PlantUmlArchCondition.java

```diff
@@ -159,20 +162,38 @@ public PlantUmlArchCondition ignoreDependencies(DescribedPredicate<? super Depen
     }
 
     @Override
-    public void check(JavaClass item, ConditionEvents events) {
-        if (allDependenciesAreIgnored(item)) {
+    public void check(JavaClass javaClass, ConditionEvents events) {
+        if (allDependenciesAreIgnored(javaClass)) {
             return;
         }
 
+        if (!javaClassDiagramAssociation.contains(javaClass)) {
+            events.add(violated(javaClass, String.format("Class %s is not contained in any component", javaClass.getName())));
+            return;
+        }
+
+        Set<PlantUmlComponent> components = javaClassDiagramAssociation.getAssociatedComponents(javaClass);
+        if (components.size() > 1) {
+            events.add(violated(javaClass, String.format(
+                    "Class %s may not be contained in more than one component, but is contained in %s",
+                    javaClass.getName(),
+                    components.stream().map(it -> it.getComponentName().asString()).sorted().collect(toList()))));
```

Comment:
Do we really want to rely on the `toString()` of the `List` built by `Collectors.toList()`?
I'd probably build the `String` explicitly:
```suggestion
            String prefix = String.format(
                    "Class %s may not be contained in more than one component, but is contained in [",
                    javaClass.getName());
            String message = components.stream().map(it -> it.getComponentName().asString()).sorted()
                    .collect(joining(", ", prefix, "]"));
            events.add(violated(javaClass, message));
```

Reply:
Hmm, I consciously did this because
a) I think the JDK's `Collection.toString()` methods have been very stable and consistent ... forever. I.e. I would trust the JDK to not suddenly change such a central API.
b) we have tests that run on each JDK version, so even if this would change we would notice it
But, on the other hand I agree that it's more explicit to read, if we do it ourselves :man_shrugging: So in the end no super strong opinion, I'll just go with your comment then :wink: (and on the other hand, the JDK did suddenly change `Annotation.toString()` in some version :joy:)